server/web/context/context.go

```diff
@@ -77,6 +78,91 @@ type Context struct {
 	_xsrfToken     string
 }
 
+type StreamEvent struct {
+	Content          string
+	ReasoningContent string
+	Error            error
+	Done             bool
+}
+
+func (ctx *Context) EventStreamResp() chan<- StreamEvent {
```

Comment:
return chan <- []byte
```go
select {
    case event, ok := dataCh:
         if !ok {
               return
         }
        // write to the resp and flush.
}
```
I think not only AI but also other scenarios may use this function.